lib/action.ml

```diff
@@ -317,25 +317,29 @@ module Action (Github_api : Api.Github) (Slack_api : Api.Slack) = struct
         Lwt.return_none
     in
     let process link =
+      let with_gh_result_populate_slack (type a) ~(api_result : (a, string) Result.t)
```

Comment:
is `type a` really needed, looks like simple `'a` will do

Reply:
oh yep, I asked @Khady about this because I couldn't get it to stay `'a` polymorphic without this.  it keeps getting type-restricted after the first use.  Louis's idea is probably because the polymorphic `'a` has a scope so it will be restricted within the `process` function (and when we lift it out of `process` it does not get restricted) so i think he's right.  I haven't found any doc relating to this though